0002-data-collection-extension/README.md

```diff
@@ -1,96 +1,129 @@
 ---
-name: Data Collection Extension protocol
+name: Data Collection extension protocol
 shortname: 2/DCX
 status: draft
 editor: Alexey Shmalko <ashmalko@cybervisiontech.com>
 contributors: Alexey Gamov <agamov@cybervisiontech.com>
 ---
 
-## Introduction
+- [Introduction](#introduction)
+- [Language](#language)
+- [Requirements and constraints](#requirements-and-constraints)
+  - [Problems and possible solutions](#problems-and-possible-solutions)
+- [Use cases](#use-cases)
+  - [UC1: individually authenticated device](#uc1-individually-authenticated-device)
+  - [UC2: mass production device](#uc2-mass-production-device)
+- [Design](#Design)
+  - [Batch uploads](#batch-uploads)
+  - [No timestamp handling](#no-timestamp-handling)
+  - [Request/response](#request-response)
+  - [Formats](#formats)
+    - [Schemeless JSON](#schemeless-json)
+      - [Request](#request)
+      - [Response](#response)
 
-The Data Collection Extension protocol is an endpoint-aware [Kaa Protocol](/0001-kaa-protocol/README.md) extension.
+## Introduction
+The Data Collection extension protocol is an endpoint-aware [Kaa Protocol](/0001-kaa-protocol/README.md) extension.
 
-It is intended to solve the problem of transfering collected data to a server for further storing or processing.
+It is designed to collect data from other services/extensions and transfer it to a server for storage and further processing.
```

Comment:
Endpoint data --> ECSKP --> DCX, right? If so, then how exactly does DCX collect data from the endpoints directly?